__tests__/endpoints.test.js

```diff
@@ -113,6 +113,96 @@ describe('GET /api/articles/:article_id', () => {
+            })
+        })
+        test("200: works for negative increments", () => {
+            const requestBody = {inc_votes: -1}
+            return request(app)
+            .patch('/api/articles/1')
+            .send(requestBody)
+            .expect(200)
+            .then(({body}) => {
+                const article = body.article
+                expect(article.article_id).toEqual(1);
+                expect(article.title).toEqual("Living in the shadow of a great man");
+                expect(article.topic).toEqual("mitch");
+                expect(article.author).toEqual("butter_bridge");
+                expect(article.created_at).toEqual(expect.any(String));
+                expect(article.votes).toEqual(99);
+                expect(article.article_img_url).toEqual(expect.any(String));
+            })
+        })
+        test('200: ignores surplus properties in the request body', () => {
+            const requestBody = {inc_votes: -1, surplus: 'property'}
+            return request(app)
+            .patch('/api/articles/1')
+            .send(requestBody)
+            .expect(200)
+            .then(({body}) => {
+                const article = body.article
+                expect(article.article_id).toEqual(1);
+                expect(article.title).toEqual("Living in the shadow of a great man");
+                expect(article.topic).toEqual("mitch");
+                expect(article.author).toEqual("butter_bridge");
+                expect(article.created_at).toEqual(expect.any(String));
+                expect(article.votes).toEqual(99);
+                expect(article.article_img_url).toEqual(expect.any(String));
+            })
+        })
+    });
+    describe('400', () => {
+        test('400: returs error for invalid inc_vote', () => {
+            const requestBody = {inc_votes: 'invalid'}
+            return request(app)
+            .patch('/api/articles/1')
+            .send(requestBody)
+            .expect(400)
+            .then(({ body }) => {
+              expect(body).toEqual({ msg: 'Bad request.' });
+            });
+        });
+        test('400: returs error for invalid :article_id', () => {
+            const requestBody = {inc_votes: -1}
+            return request(app)
+            .patch('/api/articles/forty-two')
+            .send(requestBody)
+            .expect(400)
+            .then(({ body }) => {
+              expect(body).toEqual({ msg: 'Bad request.' });
+            });
+        });
+      });
+    describe('404', () => {
+        test('404: returs error for valid but non-existent :article_id', () => {
+            const requestBody = {inc_votes: -1}
+            return request(app)
+            .patch('/api/articles/99')
+            .send(requestBody)
+            .expect(404)
+            .then(({ body }) => {
+              expect(body).toEqual({ msg: 'Resource not found' });
+            });
+        });
+    });
+});
```

Comment:
Awesome testing so far. Can we also get one for a missing inc_votes, too? That would be the last test we want here before merging